src/librustc/traits/specialize/mod.rs

```diff
@@ -366,8 +366,12 @@ pub(super) fn specialization_graph_provider(
                     }
                 }
 
-                for cause in &overlap.intercrate_ambiguity_causes {
-                    cause.add_intercrate_ambiguity_hint(&mut err);
+                let access_levels = tcx.privacy_access_levels(impl_def_id.krate);
+                let id = tcx.hir().as_local_hir_id(impl_def_id).unwrap();
```

Comment:
You should use `if let Some(id) =` instead of `unwrap()`. Every `unwrap` is a potential ICE.

Reply:
Will do, should I fix the other two `.unwrap()`s in the `specialization_graph_provider` function?

Reply:
Let's leave them for now.

Reply:
These are both fine to leave as-is (they can never fail, if they do, an ICE is appropriate). They could be turned into `.expect` with an explanation, but that's not that important (finding the source of the would-be ICEs would be easy even without).